flexget/plugins/api_tvmaze.py

```diff
@@ -0,0 +1,391 @@
+        if not db_genre:
+            db_genre = TVMazeGenre(name=genre)
+            session.add(db_genre)
+        db_genres.append(db_genre)
+    return db_genres
+
+
+def search_params_for_series(**lookup_params):
+    search_params = {
+        'tvmaze_id': lookup_params.get('maze_id') or lookup_params.get('tvmaze_id'),
+        'tvdb_id': lookup_params.get('tvdb_id'),
+        'tvrage_id': lookup_params.get('tvrage_id'),
+        'name': lookup_params.get('title') or lookup_params.get('series_name')
+    }
+    return search_params
+
+
+@with_session
+def from_cache(session=None, search_params=None, cache_type=None):
+    """
+    Returns a result from requested table based on search params
+    :param session: Current session
+    :param search_params: Relevant search params. Should match table column names
+    :param cache_type: Object for search
+    :return: Query result
+    """
+    if not any(search_params.values()):
+        raise LookupError('No parameters sent for cache lookup')
+    else:
+        log.debug('searching db {0} for the values {1}'.format(cache_type.__tablename__, search_params.items()))
+        result = session.query(cache_type).filter(
+            or_(getattr(cache_type, col) == val for col, val in search_params.iteritems() if val)).first()
+    return result
+
+
+@with_session
+def from_lookup(session=None, title=None):
+    log.debug('searching lookup table using title {0}'.format(title))
+    return session.query(TVMazeLookup).filter(func.lower(TVMazeLookup.search_name) == title.lower()).first()
+
+
+def prepare_lookup_for_pytvmaze(**lookup_params):
+    """
+    Return a dict of params which is valid with pytvmaze get_show method
+    :param lookup_params: Search parameters
+    :return: Dict of pytvmaze recognizable key words
+    """
+    prepared_params = {}
+    series_name = lookup_params.get('series_name') or lookup_params.get('show_name') or lookup_params.get('title')
+    title, year_match = split_title_year(series_name)
+
+    prepared_params['maze_id'] = lookup_params.get('tvmaze_id')
+    prepared_params['tvdb_id'] = lookup_params.get('tvdb_id') or lookup_params.get('trakt_series_tvdb_id')
+    prepared_params['tvrage_id'] = lookup_params.get('tvrage_id') or lookup_params.get('trakt_series_tvrage_id')
+    prepared_params['show_name'] = title
+    prepared_params['show_year'] = lookup_params.get('trakt_series_year') or lookup_params.get(
+        'year') or lookup_params.get('imdb_year') or year_match
+    prepared_params['show_network'] = lookup_params.get('network') or lookup_params.get('trakt_series_network')
+    prepared_params['show_country'] = lookup_params.get('country') or lookup_params.get('trakt_series_country')
+    prepared_params['show_language'] = lookup_params.get('language')
+
+    return prepared_params
+
+
+class APITVMaze(object):
+    @staticmethod
+    @with_session
+    def series_lookup(session=None, force_cache=False, **lookup_params):
```

Comment:
We use `only_cached` rather than `force_cache` in other lookup apis, I think I like that name a bit better, and to keep consistent.